### PR TITLE
fix: allow use by name alone

### LINKED ISSUE #1552: Infra use command not checking for type
```
mchiang@Michaels-MacBook-Pro infra % ./infra use ubernetes.development-72f9584e 
Switched to context "infra:development-72f9584e".
```
```
Switched to context "infra:development-72f9584e".
mchiang@Michaels-MacBook-Pro infra % ./infra use development-72f9584e  
ERROR  invalid argument
```

We should just allow the name of the cluster, and only require the type if there are duplicate names from multiple infrastructures (i.e. kubernetes cluster named dev  and a database named dev )

### PR DESCRIPTION
**Pull request by mxyng**
## Summary

`infra use` should allow referencing by name, i.e. without the `kubernetes.` prefix

## Checklist

- [ ] Wrote appropriate unit tests
- [ ] Considered security implications of the change
- [ ] Updated associated docs where necessary
- [ ] Updated associated configuration where necessary
- [ ] Change is backwards compatible if it needs to be (user can upgrade without manual steps?)
- [ ] Nothing sensitive logged
- [ ] Commit message conforms to [Conventional Commit][1]
- [ ] GitHub Actions are passing
- [ ] Considered data migrations for smooth upgrades

[1]: https://www.conventionalcommits.org/en/v1.0.0/

## Related Issues

Resolves #1552

### CODE REVIEW

File: internal/cmd/kubernetes.go [hunk 1]
```diff
@@ -23,25 +23,35 @@ func clientConfig() clientcmd.ClientConfig {
 	return clientcmd.NewNonInteractiveDeferredLoadingClientConfig(loadingRules, &clientcmd.ConfigOverrides{})
 }
 
-func kubernetesSetContext(name string) error {
+func kubernetesSetContext(cluster, namespace string) error {
```
Comment (dnephin) [inline internal/cmd/kubernetes.go:26]: Nice! I better understand how this works now!

A couple tests for this function to show it builds the expected config might be nice, but not blocking.

File: internal/cmd/cmd.go [hunk 2]
```diff
@@ -200,17 +199,22 @@ $ infra use kubernetes.development.kube-system
 				return err
 			}
 
-			parts := strings.Split(name, ".")
+			parts := strings.Split(destination, ".")
```
Comment (dnephin) [inline internal/cmd/cmd.go:202]: Seems we rely on there being at most 2 parts, so maybe this?

```suggestion
			parts := strings.SplitN(destination, ".", 2)
```

**Review by dnephin — APPROVED** [review]
Good fix! Some test coverage for `kubernetesSetContext` and/or the `infra use`  command would be great, but could be a follow up.

File: internal/cmd/cmd.go (hunk 2 above)
Comment (mxyng) [reply internal/cmd/cmd.go:202]: Ah it's possible the user supplies more than 2 parts, e.g. `kubernetes.cluster.namespace`, in which case we want to discard the first part and pass the second and third part to `SetContext`

**Pull request merged**